core/shared/src/main/scala/newts/Newtype.scala

```diff
@@ -0,0 +1,18 @@
+package newts
+
+/**
+  * `S` is a newtype means `S` is a wrapper around an `A`
+  */
+trait Newtype[S, A] {
+  def wrap(value: A): S
+  def unwrap(value: S): A
+}
```

Comment:
I feel like the `A` should be a type member here, since for some `S`, there is only one `A` that can be extracted. E.g., with `Max[Int]`, you know you will always unwrap `Int`. Nice side effect is that you can then use `@typeclass`.

And this might also be useful:
```scala
implicit class Wrap[A](self: A) {
  def wrap[S](implicit S: Newtype.Aux[S, A]) = S.wrap(self)
}
```